p2p/host/basic/basic_host.go

```diff
@@ -709,6 +709,11 @@ func (h *BasicHost) Connect(ctx context.Context, pi peer.AddrInfo) error {
 		}
 	}
 
+	addrs := h.Peerstore().Addrs(pi.ID)
+	if len(addrs) == 0 {
```

Comment:
On second thought we already return a similar error here https://github.com/libp2p/go-libp2p/blob/master/p2p/net/swarm/swarm_dial.go#L340. So this change is not required. Sorry I added confusion here.

Reply:
that's fair. Thanks for fixing this!